server/api/student/student.controller.js

```diff
@@ -84,6 +87,91 @@ exports.outreachCounts = function(req, res) {
   });
 };
 
+/**
+ * Get summary of outreaches.
+ * restriction: 'teacher'
+ *
+ * Returns an aggregation for records based on the req user role:
+ * - teachers will get outreach summary for assignment school
+ * - manager+ will get outreach summary for all schools
+ */
+exports.outreachSummary = function(req, res) {
+  var pipeline = [{
+   $group: {
+     _id: {student: '$student', type: '$type'},
+     count: {$sum: 1}
+   }
+  }, {
+   $group: {
+     _id: '$_id.student',
+     counts: {$push: {type: '$_id.type', count: '$count'}}
+   }
+  }, {
+    $project: {
+      _id: 0,
+      student: '$_id',
+      counts: 1
+    }
+  }];
+  if (req.user.role === 'teacher') {
+    pipeline.unshift({
+       $match: {school: req.user.assignment}
+    });
+  }
+  Outreach.aggregate(pipeline, function(err, results) {
+    if (err) handleError(res, err);
+    Student.populate(results, {path: 'student'})
```

Comment:
Actually since we don't need the promise we probably want to just call the function with the callback:

```
  Outreach.aggregate(pipeline, function(err, results) {
    if (err) handleError(res, err);
    Outreach.populate(results, {
      path: 'student',
      model: 'Student',
      select: 'firstName lastName studentId currentSchool',
      populate: {
        path: 'currentSchool',
        model: 'School',
        select: 'name'
      }
    }, function(err, final) {
      if (err) return handleError(res, err);
      return res.status(200).json(final);
    });
  });
```